modules/core_project_factory/main.tf

```diff
@@ -214,7 +214,7 @@ resource "google_service_account_iam_member" "service_account_grant_to_group" {
   Account on shared VPC
  *****************************************************************************************************************/
 resource "google_project_iam_member" "controlling_group_vpc_membership" {
-  count = "${(var.shared_vpc != "" && (length(compact(var.shared_vpc_subnets)) > 0)) ? local.shared_vpc_users_length : 0}"
+  count = "${(var.shared_vpc != "" && (length(compact(var.shared_vpc_subnets)) == 0)) ? local.shared_vpc_users_length : 0}"
```

Comment:
I thought this was a bug - is the intent of this behavior to grant full access to the shared VPC subnets if no subnets are specifically provided? The existing code and documentation doesn't make this behavior clear.

Reply:
#97 explains the expectations.

Reply:
Agreed, we should add some tests for this.